src/main/java/edu/st/client/controllers/SingleGameController.java

```diff
@@ -0,0 +1,190 @@
+package edu.st.client.controllers;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+import edu.st.client.models.Player;
+import edu.st.client.services.FxService;
+import edu.st.client.services.GameService;
+import edu.st.common.Util;
+import edu.st.common.models.Token;
+import javafx.fxml.FXML;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.HBox;
+
+public class SingleGameController extends BaseController {
+  public AnchorPane profile1 = null;
+  public AnchorPane profile2 = null;
+  public Button restart_btn = null;
+  public Button return_btn = null;
+  public AnchorPane overlay = null;
+  public GridPane grid = null;
+
+  public SingleGameController(Player player, Player cpu) {
+    this.player = player;
+    this.cpu = cpu;
+    initBoard();
+    // FxService.setMedia("audio/vs.mp3");
+    // FxService.playMedia();
+  }
+
+  @FXML
+  public void initialize() {
+    GameService.setPlayerProfile(profile1, player);
+    GameService.setPlayerProfile(profile2, cpu);
+
+    restart_btn.setOnAction(event -> {
+      overlay.setVisible(false);
+      resetGame();
+    });
+
+    return_btn.setOnAction(event -> {
+      FxService.switchViews("views/Login.fxml", null);
+    });
+  }
+
+  @FXML
+  public void playerClickedTile(MouseEvent event) {
+    if (currentPlayer == Token.Y) {
+      return;
+    }
+
+    Node node = event.getPickResult().getIntersectedNode();
+    // clicked on anything else but inner tile
+    if (!(node instanceof HBox)) {
+      return;
+    }
+
+    HBox tile = (HBox) node;
+    Integer col = GridPane.getColumnIndex(tile);
+    Integer row = GridPane.getRowIndex(tile);
+
+    // not a valid index
+    if (col == null && row == null) {
+      return;
+    }
+
+    // tile is already taken
+    if (board.get(row).get(col) != null) {
+      return;
+    }
+
+    updateBoard(tile, row, col);
+
+    if (Util.isWinner(board) || Util.isBoardFull(board)) {
+      GameService.addTask(700, () -> {
+        overlay.setVisible(true);
+      });
+    } else {
+      GameService.addTask(700, () -> {
+        cpuMove();
+        if (Util.isWinner(board) || Util.isBoardFull(board)) {
```

Comment:
can we make this check in cpuMove one more time just to insure we dont ever have an infinite loop. instead of overlay true we return